File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l=0
        r=len(nums)-1
        sm=nums[l]

        while l<=r:
            if nums[l]<nums[r]:
                sm=min(sm,nums[l])
                break
            mid=(l+r)//2
            sm=min(nums[mid],sm)
            
            if nums[mid]<nums[l]:
                r=mid-1
            else:
                l=mid+1
        return sm

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int min=nums[0];
        int l=0,r=nums.length-1;

        while(l<=r){
            if(nums[l]<nums[r]){
                min=Math.min(min,nums[l]);
                break;
            }

            int mid=(l+r)/2;
            min=Math.min(min,nums[mid]);
            //System.out.println(nums[mid]+","+min);

            if(nums[mid]>=nums[l]){
                l=mid+1;
                System.out.println("hi");
            }
            else{
                r=mid-1;
            }
        }
        return min;
    }
}